src/access/RequiredPrivilege.cpp

```diff
@@ -0,0 +1,214 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "RequiredPrivilege.h"
+
+namespace {
+
+using namespace chip;
+using namespace chip::Access;
+
+// Privilege override entries are stored in a table per operation (read attribute,
+// write attribute, invoke command, read entry). Entries in each table are sorted
+// by their primary compound key of cluster/endpoint/field, and therefore cannot
+// have duplicate privilege for any such combination. Cluster cannot be invalid,
+// so that is used as a sentinel for empty/unused entries, which are kept at the
+// end of the table. Endpoint and field can be invalid, which means wildcard.
+// Sorting the entries in this way allows a cluster to override specific endpoints
+// or fields with privilege, while still using wildcard to override all the
+// unspecified ones.
+struct PrivilegeOverride
+{
+    ClusterId mCluster;
+    EndpointId mEndpoint;
+    FieldId mField;
+    Privilege mPrivilege;
```

Comment:
Not sure whether we care, if this is temporary, but I expect that putting mPrivilege next to mEndpoint will make this struct 12 bytes instead of 16 on arm32.

Reply:
Thanks I did wonder about packing but just didn't do it last evening while trying to post this straw-man PR.

Reply:
Done.